protobuf-ckb-syscalls/src/lib.rs

```diff
@@ -175,14 +179,28 @@ impl SyscallImpls for ProtobufBasedSyscallImpls {
 
     fn load_cell_code(
         &self,
-        _buf_ptr: *mut u8,
-        _len: usize,
+        buf_ptr: *mut u8,
+        len: usize,
         _content_offset: usize,
         _content_size: usize,
         _index: usize,
         _source: Source,
     ) -> Result<(), Error> {
-        panic!("Load cell data as code is not suported!");
+        match self.syscall() {
```

Comment:
I recommend that a `#[cfg(target_arch = "riscv64")]` guard shall be added here. In native environment(such as fuzzing), this implementation does not really work.

Reply:
To me this is not a proper change.

The point of having `CkbFlavoredImplSyscalls`, is that there might well be other syscall implementations that could take advantage of `CkbFlavoredImplSyscalls`. So only either of the following 2 is IMHO a good solution:

* Keep the generic logic in `CkbFlavoredImplSyscalls` in fuzzing utils crate, and keep `ProtobufBasedSyscalls` in protobuf-ckb-syscalls crate
* We remove `CkbFlavoredImplSyscalls` and only keep `ProtobufBasedSyscalls`

So the question here is: can we build a generic enough type `CkbFlavoredImplSyscalls` that has all the details of `load_cell_code`, but allow customization of how to fetch the actual code data? If so, we should split the 2 and keep them in separate packages, otherwise, we should only have `ProtobufBasedSyscalls`

Reply:
> I believe the key to implementing it is a way for load_cell_code to proactively send information to the outside world. This could be achieved through return values or callbacks, but we've already proven that this isn't a good approach.

Note that solution 1 does not necessarily mean adjusting the signature of `load_cell_code`, those are 2 different issues. Solution 1 is above making `CkbFlavoredImplSyscalls` a trait, and expose a overridable method so in `ProtobufBasedSyscalls` we can fill in the missing part by returning code data.

The fact that returning values in `load_cell_code` is a bad idea, does not necessarily mean the same idea cannot be applied on a different level.

Reply:
The second solution seems more appealing to me.

For the first idea, I don't have any particularly good suggestions for improvement. Intuitively, my thought is to add a flag to `ProtobufBasedSyscallImpls` to indicate whether `load_cell_code` should directly panic or perform an actual memory copy. This way, we can keep the original versions of `CkbFlavoredImplSyscalls` and `ProtobufBasedSyscall` unchanged while supporting both fuzzing and debugger use cases. However, this modification feels quite crude.

Reply:
As for the design of CkbFlavoredImplSyscalls, I think it's still an open topic, perhaps better discussed in a new issue rather than pull requests.

I've submitted a new commit following the solution 2: "We remove CkbFlavoredImplSyscalls and only keep ProtobufBasedSyscalls."This choice assumes that both solutions are "good solution".